packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActionButton,
+  DialogActions,
+  DialogCancelButton,
+  DialogContent,
+  DialogHeader,
+  Form,
+  FormInput,
+  Text,
+} from "@prismicio/editor-ui";
+import { useState } from "react";
+
+import {
+  addUIDField,
+  getUIDField,
+  getUIDFieldLabel,
+  updateUIDField,
+} from "@/domain/customType";
+import { useCustomTypeState } from "@/features/customTypes/customTypesBuilder/CustomTypeProvider";
+
+export function UIDEditor() {
+  const [isOpen, setOpen] = useState(false);
+  const { customType, setCustomType } = useCustomTypeState();
+  const field = getUIDField(customType);
+  const uidFieldLabel = getUIDFieldLabel(customType);
+  const [label, setLabel] = useState(uidFieldLabel ?? "");
+
+  function handleSubmit() {
+    if (!label) {
+      return;
+    }
+    const updatedCustomType = field
+      ? updateUIDField(label, customType)
+      : addUIDField(label, customType);
+
+    setCustomType(updatedCustomType);
+    setOpen(false);
+  }
+
+  return (
+    <Dialog
+      open={isOpen}
+      onOpenChange={setOpen}
+      size="small"
+      trigger={
+        <Button
+          color="grey"
+          textColor="placeholder"
+          startIcon={field ? "language" : "add"}
+          sx={{ marginInline: "auto" }}
+        >
+          {field ? uidFieldLabel : "Add an UID"}
+        </Button>
+      }
+    >
+      <DialogHeader title="Edit the UID label" />
+      <DialogContent>
+        <Form onSubmit={handleSubmit}>
+          <Box flexDirection="column" padding={16} gap={4}>
+            <FormInput
+              type="text"
+              label="Label *"
+              placeholder="UID"
+              value={label}
+              onValueChange={setLabel}
+              error={!label && "This field is required"}
+            />
+            {label && <Text color="grey11">A label for the UID</Text>}
```

Comment:
⚠️ #issue

TODO just to remember to use latest editor package when ready 🙂

Reply:
You mean that this text would be passed directly to FormInput with new package, right?
I'm not 100% sure if this change will be applied in the FormInput, as far as I know there is still some discussion about that. 
But I can add comment to update the code if the change is made in editor-ui.

Reply:
Let's ask José, I think something is still in progress.

Reply:
yeah, it is still not unclear, will leave comment like that:
`{/* TODO: refactor if change proposed in: https://github.com/prismicio/editor/pull/1151 is released */}`